lib/rucio/daemons/replicarecoverer/suspicious_replica_recoverer.py

```diff
@@ -285,48 +287,50 @@ def declare_suspicious_replicas_bad(once=False, younger_than=3, nattempts=10, vo
                     # Get the rse_id by going to one of the suspicious replicas from that RSE and reading it from there
                     rse_id = list(recoverable_replicas[vo][rse_key].values())[0]['rse_id']
                     for replica_key in list(recoverable_replicas[vo][rse_key].keys()):
+                        file_metadata = get_metadata(recoverable_replicas[vo][rse_key][replica_key]["scope"], recoverable_replicas[vo][rse_key][replica_key]["name"])
+                        recoverable_replicas[vo][rse_key][replica_key]["datatype"] = str(file_metadata["datatype"])
                         if recoverable_replicas[vo][rse_key][replica_key]['available_elsewhere'] is True:
                             # Replicas with other copies on at least one other RSE can safely be labeled as bad
-                            files_to_be_declared_bad.append(recoverable_replicas[vo][rse_key][replica_key]['surl'])
+                            files_to_be_declared_bad.append(recoverable_replicas[vo][rse_key][replica_key])
```

Comment:
Was this a bug?

Reply:
No, but we are changing the log messages for files that are ignored or declared bad (line 324, 327) so that we have more information in the log file about which replicas are being affected. For this change to work, we need more information than just the surl of each file.

Reply:
Okay, seems reasonable. This is also used in the `pfns`, which will be passed to the `declare_bad_file_replicas`. I think this should be fine.